.github/workflows/js_tests.yml

```diff
@@ -0,0 +1,272 @@
+name: JS tests master
+
+on:
+  push:
+    branches:
+      - master
+      - 3.x-maintenance
+    paths:
+      - '.github/scripts/*.sh'
+      - '.github/workflows/*.yml'
+      - 'gui/package.json'
+      - 'requirements/*.txt'
+      - 'Makefile'
+      - 'MANIFEST.in'
+      - '**.bat'
+      - '**.jsx?'
+      - '**.sh'
+
+  pull_request:
+    branches:
+      - master
+      - 3.x-maintenance
+    paths:
+      - '.github/scripts/*.sh'
+      - '.github/workflows/*.yml'
+      - 'gui/package.json'
+      - 'requirements/*.txt'
+      - 'MANIFEST.in'
+      - '**.bat'
+      - '**.jsx?'
+      - '**.sh'
+
+jobs:
+  ############################################################################
+  # Close previous builds on the same branch
+  ############################################################################
+  cleanup-previous-runs:
+    name: Cleanup previous builds
+    runs-on: ubuntu-latest
+    steps:
+      - run: echo $GITHUB_EVENT_NAME
+      - uses: rokroskar/workflow-run-cleanup-action@v0.2.2
+        if: github.event_name == 'pull_request'
+        env:
+          GITHUB_TOKEN: "${{ secrets.GITHUB_TOKEN }}"
```

Comment:
Will this even be available when pull requests run? If it is, is there any risk this can leak?

Reply:
Dont know

Reply:
run gui/package.json in CI since it has been broken for ages.

Reply:
I'm not sure how to interpret that actually. It doesn't explicitly say anything as far as I can tell. I tried googling yesterday as well without success.

The attack vector I want  to know is: can someone make a PR that adds/modifies a workflow and then snag the access token?

Reply:
https://github.community/t5/GitHub-Actions/Token-permissions-for-forks-once-again/td-p/33839

from forks I think not...?

Reply:
If it's not run for forks, then this only works when one of us maintainers does a PR directly on the core lektor repository I guess? Is that case worth optimizing for?

Reply:
>If it's not run for forks, then this only works when one of us maintainers does a PR directly on the core lektor repository I guess? Is that case worth optimizing for?

Good question I need to check this